model/githubapp/github_app_auth_db.go

```diff
@@ -49,28 +48,201 @@ func GetGitHubAppID(projectId string) (*int64, error) {
 	return &githubAppAuth.AppID, err
 }
 
-// UpsertGithubAppAuth inserts or updates the app auth for the given project id in the database
-func UpsertGithubAppAuth(githubAppAuth *GithubAppAuth) error {
+// githubAppCheckAndRunParameterStoreOp checks if Parameter Store is
+// enabled and if so, runs the given Parameter Store operation.
+func githubAppCheckAndRunParameterStoreOp(ctx context.Context, op func() error) error {
+	flags, err := evergreen.GetServiceFlags(ctx)
+	if err != nil {
+		return errors.Wrap(err, "getting service flags")
+	}
+	if flags.ParameterStoreDisabled {
+		return nil
+	}
+
+	return op()
+}
+
+// defaultParameterStoreAccessTimeout is the default timeout for accessing
+// Parameter Store. In general, the context timeout should prefer to be
+// inherited from a higher-level context (e.g. a REST request's context), so
+// this timeout should only be used as a last resort if the context cannot
+// easily be passed down.
+const defaultParameterStoreAccessTimeout = 30 * time.Second
+
+// FindOneGitHubAppAuth finds the GitHub app auth and retrieves the private key
+// from Parameter Store if enabled.
+func FindOneGitHubAppAuth(id string) (*GithubAppAuth, error) {
```

Comment:
Could we pass in the context here and used instead of context.Background()? If not, could we comment somewhere why we can't?

Reply:
This is mostly like this because the GitHub app logic was implemented to not pass a context. Most DB functions that use the legacy DB helpers in the `db` package haven't been updated to use contexts. I filed DEVPROD-14162 to address the tech debt.